src/plugins/data/public/ui/dataset_selector/dataset_selector.tsx

```diff
@@ -99,22 +99,41 @@ export const DatasetSelector = ({
         prepend: <EuiIcon type={datasetService.getType(type)!.meta.icon.type} />,
       });
     });
-
+    const recentlySelected: EuiSelectableOption[] = [
+      {
+        label: 'Recently Selected Data',
+        isGroupLabel: true,
+      },
+    ];
+    datasetService.getRecentDatasets().forEach(({ id, title, type, dataSource }) => {
+      const label = dataSource ? `${dataSource.title}::${title}` : title;
+      recentlySelected.push({
+        label,
+        checked: id === selectedDataset?.id ? 'on' : undefined,
+        key: id,
+        prepend: <EuiIcon type={datasetService.getType(type)!.meta.icon.type} />,
+      });
+    });
+    if (recentlySelected.length > 1) {
+      newOptions.unshift(...recentlySelected);
```

Comment:
Nit: Cant these simply be appended before index patterns?

Reply:
+1. 

also do we know if we want to display this drop down with two checkmarks? or if the recents contain a dataset id then we won't display the dataset within the original type?

For example, if they have an index pattern like `sample_data_logs` and they select it. then `sample_data_logs` will display only in the recents section with a section and the index patterns section will not have `sample_data_logs`.

or did we want two checkmarks with `sample_data_logs` existing in both places?

Reply:
Thats a good callout. For simplicity I think we should deduplicate it and show it only in the recents. That way showing the check mark will also be easier